xmpp.go

```diff
@@ -664,7 +667,7 @@ func (c *Client) init(o *Options) error {
 			case slices.Contains(mechSlice, "X-OAUTH2"):
 				mechanism = "X-OAUTH2"
 				// Do not use PLAIN auth if NoPlain is set.
-			case slices.Contains(mechSlice, "PLAIN") && tlsConnOK && !o.NoPLAIN:
+			case slices.Contains(mechSlice, "PLAIN") && (tlsConnOK || o.NoTLS) && !o.NoPLAIN:
```

Comment:
I think from a security point of view we should make it *very* hard to use PLAIN auth over an unencrypted connection. So maybe better not allow PLAIN here, but set set `Mechanism` in `Options` to `PLAIN`, then the else clause won't be used.